rfcs/rfc-bld-20230302-1-3p-development.md

```diff
@@ -0,0 +1,215 @@
+
+$~$
+
+### 3P Package Contribution Phase
+
+1.  When the contribution is ready, a 3P package PR will be created by the contributor, targeting the 3p-package-source development branch  
+    1.  Prechecks via Github Actions will get triggered on PR creation to validate the **`PackageInfo.json`** file for completeness and the license indicated in the file is "blessed" by the project.
+    2.  Recursive dependencies and licenses **should** be called out, but a secondary license check using [Scancode Toolkit](https://github.com/nexB/scancode-toolkit) may be needed to validate
+2.  A maintainer will manually execute the build step in Github Actions once all prechecks are green. An automated comment referencing a codeowners approval group can be added to the PR, which would notify maintainers.  
+    
+3.  If the build fails, the 3P Package PR is kicked back for re-work
+4.  Once the build step is complete, the package is automatically uploaded to Github packages for the O3DE org. The 3P package PR should be merged into 3p-package-source.
+5.  A download location and hash of the package will be generated in the build output. The package is now staged for consumption in a Pull Request for O3DE AR, but not ready for use in production
+    1.  Note: The hash generated for the O3DE staged package will not match the developer fork package
+
+$~$
+
+### O3DE Contribution/AR Phase
+
+1.  The package hash generated from the 3P package contribution phase will be used in a PR targeting O3DE's `**cmake/3rdParty/Platform/<platform>/BuiltInPackages_<platform>.cmake**` file for the staged package  
+    1.  This can be manual initially, but automated PR can be triggered immediately after the 3P Package Contribution phase once there is enough confidence in the workflow
+2.  Run the O3DE PR against AR for the new package hash and optionally, any engine changes
+3.  If the AR fails, then package changes may need to be made, kicking the workflow back to the 3P Package Development phase for a new PR iteration of the package.
+4.  If AR succeeds, then the PR is merged into O3DE development, creating a new reference to the 3P package revision in `**BuiltInPackages_<platform>.cmake**` in the development branch and "promoting" the 3p package to production
+
+$~$
+
+### Cleanup Phase
+
+1.  Development packages that were not promoted past **x** date will be subject to cleanup through a scheduled Github Action, with notification
+2.  Package versions in production can be kept up to **n** versions or indefinitely, depending on the need to keep past release versions
+
+$~$
+
+## User/Contributor workflow
+
+### On the user and AR side
+
+1.  The 3p cmake script will download the GitHub Package via Curl and unpack the Docker "image", creating the expected 3p tar.gz file. This will need to match up with the 3p packages hash file

```

Comment:
Are there any changes we need to make to the cmake scripts when we update the endpoint from Cloudfront/S3 to GitHub's package distribution?